examples/stripe-script/toolpad/resources/functions.ts

```diff
@@ -0,0 +1,52 @@
+import fs from 'fs';
+import { exec } from 'child_process';
+import path from 'path';
+import Stripe from 'stripe';
+
+const initStripe = () => {
+  if (!process.env.STRIPE_TOKEN) {
+    throw new Error('Missing STRIPE_TOKEN environment variable');
+  }
+  return new Stripe(process.env.STRIPE_TOKEN, {
+    apiVersion: '2022-11-15',
+  });
+};
+
+// const stripe = new Stripe(process.env.STRIPE_TOKEN, {
+//   apiVersion: '2022-11-15',
+// });
+
+export async function listInvoices(limit: number = 100, starting_after?: string) {
+  const stripe = initStripe();
+  const list = await stripe.invoices.list({
+    limit,
+    starting_after,
+  });
+
+  return JSON.stringify(list.data);
+}
+
+export async function downloadPDFs(limit: number = 100, starting_after?: string) {
+  const stripe = initStripe();
+
+  const list = await stripe.invoices.list({
+    limit,
+    starting_after,
+  });
+
+  list.data?.forEach((invoice) => {
+    try {
+      const dest = path.resolve('.', `invoices-${invoice.id}.pdf`);
+      // check if destination is resolvable
+      if (fs.existsSync(dest)) {
+        console.error('Invoice already exists:', dest);
+        return;
+      }
+      exec(`curl '${invoice.invoice_pdf}' -L -o '${dest}'`);
```

Comment:
An alternative to this, which works but does not give a lot of feedback, would be to create a custom component for the download button that uses an anchor tag with the `download` attribute, so that the PDF is downloaded on the browser.
Not sure how easy/possible it would be but it might be an improvement.

Reply:
The current button downloads all the invoices in the grid instead of just one, so it won't end up being an improvement, although I agree showing visual feedback during the download process is an improvement.

Personally I would like for a success toast to appear on the page once the action is complete.